analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -46,9 +46,29 @@ public abstract class SymbolicExecutionRunnerBase : SonarDiagnosticAnalyzer
             where TSonarFallback : new() =>
             new RuleFactory<TRuleCheck, TSonarFallback>();
 
-        // We need to rewrite this https://github.com/SonarSource/sonar-dotnet/issues/4824
-        protected static bool IsEnabled(SonarSyntaxNodeReportingContext context, DiagnosticDescriptor descriptor) =>
-            context.HasMatchingScope(descriptor) && descriptor.GetEffectiveSeverity(context.Compilation.Options) != ReportDiagnostic.Suppress;
+        protected static bool IsEnabled(SonarSyntaxNodeReportingContext context, DiagnosticDescriptor descriptor)
```

Comment:
This is SE-specific for performance reasons.

Normally, Roslyn is doing this magic and calls us. We don't need to do it in other rules.

Here, we have one `SymbolicExecutionRunner` that runs 11+ rules. And Roslyn will execute us in case at least one of them is active. And that has performance penalty - all of them would be passed to the on the SymbolicExecution engine and executed. And all of them would abuse the `ShouldExecute` thing, possibly running the engine everywhere everytime.
So we use this logic to remove those rules, that Roslyn already knows that should not be active.